semantic-core/src/Core/Parser.hs

```diff
@@ -61,7 +61,12 @@ application :: (TokenParsing m, Carrier sig t, Member Core sig, Monad m) => m (t
 application = projection `chainl1` (pure (Core.$$))
 
 projection :: (TokenParsing m, Carrier sig t, Member Core sig, Monad m) => m (t Name)
-projection = foldl' (Core....) <$> atom <*> many (namedValue <$ dot <*> name)
+projection = let a <$$> b = flip a <$> b in do
+  head <- atom
+  res <- many (choice [ (Core..?)  <$$> (symbol ".?" *> identifier)
+                      , (Core....) <$$> (dot *> identifier)
+                      ])
+  pure (foldr (>>>) id res head)
```

Comment:
We should really make this use `buildExpressionParser` from the `parsers` package.

Reply:
Anyway, I deemed `foldr (>>>) id res head` more reasonable than `appEndo (getDual (fold (Dual . Endo <$> res))) head`

Reply:
I don’t think I understand this fold at all; in particular, it’s not obvious to me that it’s left-associating the chain correctly, and that’s not something I’d want to have to puzzle over. Can we try something like this instead?

```haskell
foldl' (&) <$> atom <*> many (choice …)
```

Reply:
This is indeed nicer, so I’ll use it. Very impressive.